subgraphs/opensea-v2/src/utils.ts

```diff
@@ -0,0 +1,149 @@
+import { Address, BigInt, Bytes, ethereum, log } from "@graphprotocol/graph-ts";
+import {
+  BIGINT_ONE,
+  ETHABI_DECODE_PREFIX,
+  MATCH_ERC115_SAFE_TRANSFER_FROM_SELCTOR,
+  MATCH_ERC721_SAFE_TRANSFER_FROM_SELCTOR,
+  MATCH_ERC721_TRANSFER_FROM_SELCTOR,
+} from "./constants";
+
+export class DecodedTransferResult {
+  method: string;
+  from: Address;
+  to: Address;
+  token: Address;
+  tokenId: BigInt;
+  amount: BigInt;
+
+  constructor(
+    _method: string,
+    _from: Address,
+    _to: Address,
+    _token: Address,
+    _tokenId: BigInt,
+    _amount: BigInt
+  ) {
+    this.method = _method;
+    this.from = _from;
+    this.to = _to;
+    this.token = _token;
+    this.tokenId = _tokenId;
+    this.amount = _amount;
+  }
+}
+
+export function getFunctionSelector(callData: Bytes): string {
+  return Bytes.fromUint8Array(callData.subarray(0, 4)).toHexString();
+}
+
+export function checkCallDataFunctionSelector(callData: Bytes): boolean {
+  let functionSelector = getFunctionSelector(callData);
+  return (
+    functionSelector == MATCH_ERC721_TRANSFER_FROM_SELCTOR ||
+    functionSelector == MATCH_ERC721_SAFE_TRANSFER_FROM_SELCTOR ||
+    functionSelector == MATCH_ERC115_SAFE_TRANSFER_FROM_SELCTOR
+  );
+}
+
+export function guardedArrayReplace(
```

Comment:
Should add a comment with a link to https://github.com/ProjectWyvern/wyvern-ethereum/blob/bfca101b2407e4938398fccd8d1c485394db7e01/contracts/exchange/Exchange.sol#L20